### PR TITLE
[Leo Issue 215] Reduce time for cluster create request

### PR DESCRIPTION
**Pull request by vkumra-broad**
**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

After merging, _even if you haven't bumped the version_: 
- [ ] Update `README.md` and the `CHANGELOG.md` for any libs you changed with the new dependency string. The git hash is the same short version you see in GitHub (i.e. seven characters). You can commit these changes straight to develop.

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala [hunk 1]
```diff
@@ -300,4 +301,22 @@ class HttpGoogleStorageDAO(appName: String,
     }
   }
 
+  override def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName, readerEntity: List[GcsEntity], ownerEntity: List[GcsEntity]): Future[GcsBucketName] = {
```
Comment (jmthibault79) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala]: Move up in file to other `createBucket()`

**Comment by jmthibault79** [comment]
Looks good.  Add a note about this in the google CHANGELOG and replace the versions in that file with `TRAVIS-REPLACE-ME`

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala [hunk 2]
```diff
@@ -11,6 +11,7 @@ import scala.concurrent.Future
 trait GoogleStorageDAO {
 
   def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName): Future[GcsBucketName]
+  def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName, readerEntity: List[GcsEntity], ownerEntity: List[GcsEntity]): Future[GcsBucketName]
```
Comment (helgridly) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala]: just looking at the signatures: can we replace the 2-param function with default values on the 4-param?

Comment (rtitle) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala]: +1, can probably write one in terms of the other:
```
def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName): Future[GcsBucketName] = {
  createBucket(billingProject, bucketName, List.empty, List.empty)
}
```

Comment (rtitle) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala]: Or even get rid of the other one and write this as:

```
def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName, readerEntity: List[GcsEntity] = List.empty, ownerEntity: List[GcsEntity] = List.empty): Future[GcsBucketName]
```

Comment (rtitle) [reply google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala]: Also regarding variable naming, maybe `readerEntity --> readers` and `ownerEntity --> owners` would be more clear.

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package org.broadinstitute.dsde.workbench.google
 
+import java._
```
Comment (rtitle) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala]: ? I suspect this is not used.

**Review by rtitle — CHANGES_REQUESTED** [review]
Just for removing code duplication of the 2 `createBucket` methods

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala [hunk 4]
```diff
@@ -10,7 +10,7 @@ import scala.concurrent.Future
 
 trait GoogleStorageDAO {
 
-  def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName): Future[GcsBucketName]
+  def createBucket(billingProject: GoogleProject, bucketName: GcsBucketName, reader: List[GcsEntity] = List.empty, owner: List[GcsEntity] = List.empty): Future[GcsBucketName]
```
Comment (jmthibault79) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleStorageDAO.scala]: Pluralize: readers, owners

**Review by rtitle — APPROVED** [review]

**Pull request merged**